viewing_party/party.py

```diff
@@ -0,0 +1,192 @@
+    ratings = 0
+    user_watched = user_data["watched"]
+
+    if user_watched == []:
+        return float(0)
+
+    else:
+        for movie_entry in user_watched:
+            ratings += movie_entry["rating"] 
+
+    return(ratings / len(user_watched))
+
+def get_most_watched_genre(user_data):
+    """
+    Builds a frequency dict to determine a user's most watched genre according to their
+    watched list, and returns the value as a string. Does not account for ties.
+    """
+    genre_frequencies = {}
+    user_watched = user_data["watched"]
+    
+    # Counts the occurrences of each genre in the user's watched list in a dictionary, where the dict keys are the genres
+    for movie_entry in user_watched:
+        if movie_entry["genre"] not in genre_frequencies:
+            genre_frequencies[movie_entry["genre"]] = 1
+        else:
+            genre_frequencies[movie_entry["genre"]] += 1
+
+    # Iterate through genre frequency dictionary to determine largest value. Uses a counter to compare genre frequencies
+    current_most_watched_genre = 0
+    for genre, frequency in genre_frequencies.items():
+        if frequency > current_most_watched_genre:
+            current_most_watched_genre = frequency
+            
+    for genre, frequency in genre_frequencies.items(): # this could be refactored to not use two loops… 🤠
+        if frequency == current_most_watched_genre:
+            return genre
+
+# ——— WAVE 03 ———
+def get_unique_watched(user_data):
+    """
+    Returns a list of dictionaries that represents a list of movies that are in the user's watched list, 
+    but are not in any of the user's friends watched lists. The movies are unique to the user.
+    """
+    not_unique_titles = []
+    user_watched = user_data["watched"] 
+    friends_watched = user_data["friends"]
+
+    for movie in range(len(user_watched)):
+        for x in range(len(friends_watched)): 
+            for item in friends_watched[x]["watched"]:
+                if user_watched[movie]["title"] in item.values():
+                    not_unique_titles.append(item)
+
+    for title in user_watched:
+        for movie in not_unique_titles:
+            if movie in user_watched:
+                user_watched.remove(movie)
+    return user_watched
+
+def get_friends_unique_watched(user_data):
+    """
+    Returns a list of dictionaries that represents a list of movies that are in the user's 
+    friends watched list(s), but are not in the user's watched list. Friend(s) have watched movie, user has not.
+    """
+    friends_watched_titles = []
+    user_watched = user_data["watched"] 
+    friends_watched = user_data["friends"]
+
+    # Adds every movie in each friend's watched list to friends_watched_titles list
+    for x in range(len(friends_watched)): 
+        for item in friends_watched[x]["watched"]:
+            if item not in friends_watched_titles:  # to not duplicate a movie already added in a previous iteration
```

Comment:
Because `in` goes through the entire list, this section is essentially a triple nested loop.  It works, but there is a better way (above).